pkg/agent/multicast/multicast_route.go

```diff
@@ -0,0 +1,422 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package multicast
+
+import (
+	"fmt"
+	"net"
+
+	"syscall"
+	"time"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/interfacestore"
+
+	"antrea.io/antrea/pkg/util/runtime"
+	multicast_syscall "antrea.io/antrea/third_party/unix"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+const (
+	GroupNameIndexName = "groupName"
+	MulticastFlag      = "multicast"
+
+	igmpMsgminRetryDelay = 5 * time.Second
+	igmpMsgmaxRetryDelay = 300 * time.Second
+)
+
+func newRouteClient(nodeconfig *config.NodeConfig, ifaceStore interfacestore.InterfaceStore) (*MRouteClient, error) {
+	fd, err := createMulticastSocket()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create multicast socket")
+	}
+	VIFAllocator := newVIFAllocator(fd)
```

Comment:
local variable shouldn't be capitalized. vifAllocator

Reply:
updated